### PR TITLE
[FIX] web: fix pills resizing in weekly calendar view

### PR DESCRIPTION
**Pull request by arpr-odoo**
Steps to reproduce:
- Install planning and switch to calendar view
- try resizing the pill that is already resizable (pills that span for multiple days)

Issue:
pill resizing is not working as expected.

Cause:
the end date wasn't being calculated correctly for this.

Fix:
after this commit, the end date is calculated correctly.

task-3326281

### CODE REVIEW

**Review by MissingNoShiny — COMMENTED** [review]
LGTM, thanks for your work

**Comment by xavierbol** [comment]
Hello @arpr-odoo 

Thanks for your work. 🙂 

Can you also add a unit test in JS to reproduce the issue? 🙂 

Thanks in advance!

**Comment by xavierbol** [comment]
The runbot is red, can you check when you have time?

File: addons/web/static/tests/views/calendar/helpers.js [hunk 1]
```diff
@@ -501,6 +501,37 @@ export async function resizeEventToTime(target, eventId, dateTime) {
...
+    // Find event resizer
+    const resizer = event.querySelector(".fc-end-resizer");
+    resizer.style.display = "block";
+    resizer.style.width = "100%";
+    resizer.style.height = "1em";
+    resizer.style.bottom = "0";
+    const resizerRect = resizer.getBoundingClientRect();
+    const resizerPos = {
+        x: resizerRect.x + resizerRect.width,
+        y: resizerRect.y + resizerRect.height/2,
```
Comment (xavierbol) [inline addons/web/static/tests/views/calendar/helpers.js:520]:
```suggestion
        y: resizerRect.y + resizerRect.height / 2,
```

File: addons/web/static/tests/views/calendar/helpers.js [hunk 2]
```diff
@@ -501,6 +501,37 @@ export async function resizeEventToTime(target, eventId, dateTime) {
...
+    };
+    await triggerEventForCalendar(resizer, "mousedown", resizerPos);
+    // Find slot position
+    await scrollTo(slot, false);
+    const slotRect = slot.getBoundingClientRect();
+    const toPos = {
+        x: slotRect.x + slotRect.width/2,
+        y: slotRect.y + slotRect.height/2,
```
Comment (xavierbol) [inline addons/web/static/tests/views/calendar/helpers.js:528]:
```suggestion
        x: slotRect.x + slotRect.width / 2,
        y: slotRect.y + slotRect.height / 2,
```

File: addons/web/static/tests/views/calendar/helpers.js [hunk 3]
```diff
@@ -501,6 +501,37 @@ export async function resizeEventToTime(target, eventId, dateTime) {
...
+
+    await scrollTo(event);
+    await triggerEventForCalendar(event, "mouseenter");
+
+    // Find event resizer
+    const resizer = event.querySelector(".fc-end-resizer");
+    resizer.style.display = "block";
+    resizer.style.width = "100%";
+    resizer.style.height = "1em";
+    resizer.style.bottom = "0";
```
Comment (xavierbol) [inline addons/web/static/tests/views/calendar/helpers.js:516]: I am not sure to understand why we need to add those custom stylings.

File: addons/web/static/tests/views/calendar/calendar_view_tests.js [hunk 4]
```diff
@@ -4980,4 +4982,54 @@ QUnit.module("Views", ({ beforeEach }) => {
...
+                <calendar event_open_popup="1" date_start="start" date_stop="stop" all_day="allday" delete="0" mode="month" >
+                    <field name="stop"/>
+                </calendar>`,
+            mockRPC(route, { args, method }) {
+                if (method === "create") {
+                    assert.deepEqual(
+                        args[0],
+                        [
+                            {
+                                allday:true,
```
Comment (xavierbol) [inline addons/web/static/tests/views/calendar/calendar_view_tests.js:5002]:
```suggestion
                                allday: true,
```

File: addons/web/static/tests/views/calendar/calendar_view_tests.js [hunk 5]
```diff
@@ -4980,4 +4982,54 @@ QUnit.module("Views", ({ beforeEach }) => {
...
+        await selectAllDayRange(target, "2016-12-25", "2016-12-28");
+        await editInput(target, ".o-calendar-quick-create--input", "new event");
+        await click(target, ".o-calendar-quick-create--create-btn");
+        await resizeEventToDate(target, 8, "2016-12-31");
+        await clickEvent(target, 8);
+        await advanceTime(300);
+        assert.strictEqual(
+            target.querySelector(
+                ".o_cw_popover .o_cw_popover_fields_secondary .list-group-item .o_field_datetime"
+                ).textContent.split(' ')[0],
```
Comment (xavierbol) [inline addons/web/static/tests/views/calendar/calendar_view_tests.js:5030]: According to the web tooling (https://github.com/odoo/odoo/blob/62cbb2073922db98df380eacce0b620e1dda9877/addons/web/tooling/enable.sh). You can install that tooling by executing that file in your terminal.

```suggestion
                ).textContent.split(" ")[0],
```

File: addons/web/static/tests/views/calendar/calendar_view_tests.js [hunk 6]
```diff
@@ -4980,4 +4982,54 @@ QUnit.module("Views", ({ beforeEach }) => {
...
+                                allday:true,
+                                name: "new event",
+                                start: "2016-12-25",
+                                stop: "2016-12-28",
+                            },
+                        ],
+                        "should send the correct data to create events"
+                    );
+                }
+                if (method === "write") {
```
Comment (xavierbol) [inline addons/web/static/tests/views/calendar/calendar_view_tests.js:5011]:
```suggestion
                } else if (method === "write") {
```

**Review by xavierbol — COMMENTED** [review]
Thanks again for your work. 🙂 

I left some comments. 🙂

File: addons/web/static/tests/views/calendar/helpers.js (hunk 3 above)
Comment (arpr-odoo) [reply addons/web/static/tests/views/calendar/helpers.js:516]: Hello @xavierbol 

We're not giving styling to `pill` but we're giving styling to `.fc-end-resizer`
which won't be visible in the UI until we hover at the end of the pill.
now the thing is by giving height and width we're making sure that
`.fc-end-resizer` is at the end of a pill which will be `resizerPos`.

(we can do this even by reducing/adding a few degrees from "X" and "Y" of `resizerPos` but wouldn't it
be inconvenient to do? :thinking: )

removing this styling will lead to an Error where "mousedown won't be able to find the correct 
target".
let me know if there is a better solution to this that I'm not aware of. :thinking: 
Thanks :slightly_smiling_face:

**Comment by xavierbol** [comment]
Thanks for your work, I will let the framework JS do the final review. 🙂

**Comment by xavierbol** [comment]
Hello @mcm-odoo 

Can we have your opinion when you have time?

(I retarget the fix since the 16.4 is no longer maintain.)

**Review by mcm-odoo — APPROVED** [review]
I was a bit scared about [this change](https://github.com/odoo/odoo/pull/139893/files#diff-af6f34ca17bbd10794755dc9a85abe3f4d953a70cb2486d67730d1a536502d48R157) but it looks like it works in other apps so LGTM.

**Review by xavierbol — APPROVED** [review]
robodoo r+

**Pull request closed without merging**